src/websocket_broker/ws_handle.go

```diff
@@ -0,0 +1,165 @@
+package websocket_broker
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"sync"
+
+	"github.com/HyperloopUPV-H8/Backend-H8/common"
+	"github.com/HyperloopUPV-H8/Backend-H8/websocket_broker/models"
+	"github.com/google/uuid"
+	"github.com/gorilla/websocket"
+)
+
+var (
+	upgrader websocket.Upgrader = websocket.Upgrader{
+		CheckOrigin: func(*http.Request) bool { return true },
+	}
+	broker *WebSocketBroker = nil
+)
+
+func Get() *WebSocketBroker {
+	if broker == nil {
+		initBroker()
+	}
+	return broker
+}
+
+func initBroker() {
+	broker = &WebSocketBroker{
+		handlers:   make(map[string][]models.MessageHandler),
+		handlersMx: &sync.Mutex{},
+		clients:    make(map[string]*websocket.Conn),
+		clientsMx:  &sync.Mutex{},
+	}
+}
+
+type WebSocketBroker struct {
+	handlers   map[string][]models.MessageHandler
+	handlersMx *sync.Mutex
+	clients    map[string]*websocket.Conn
+	clientsMx  *sync.Mutex
+}
+
+func (broker *WebSocketBroker) HandleConn(writter http.ResponseWriter, request *http.Request) {
+	defer request.Body.Close()
```

Comment:
for example, move this logic into a new WebSockerConnectionHandler file

Reply:
The code is better understood if everything is on the same file, and the file is not too long

Reply:
Precisely, having everything in one file makes the code more difficult to read, difficult to understand and prevents it from being scalable.

Like everything I usually comment on here, it does not mean that the solution you choose is wrong, of course, I simply leave tips that will help to make a more professional and robust code.

I can buy the argument of "it's not worth changing this" because in the end you know how much time you have and many times I'm "picky" with this kind of thing, but my goal is that if you leave the code like that, it's because you've decided to do it consciously and considering the cons, and not because you have not known how to do it in another way.

That being said, it is true that a 150-200 line file is not excessively large, but in this case, there is a lot of extractable logic being written here. The fact that a method is extracted does not mean that this method no longer exists in this file, the inner logic is simply moved to an external file and you access it from the same current function of this file . This makes it easier to read because, for example, if you are looking for a bug in the HandleConn function, your file will have little content and you will be able to see very easily that the logic with which you have problems is in its own file that takes care of that. So the rest of the logic of the other functions does not interfere with your true objective.

Reply:
and a clear symptom of this is that the code has many indentations and you make use of breaks and empty returns in the middle of loops